### PR TITLE
Update VM Extension doc and improve README.md

### PR DESCRIPTION
**Pull request by lagalbra**
@Microsoft/omsagent-devs 

VM-Extension.md has been updated from the latest changes [here](https://github.com/Azure/azure-linux-extensions/blob/master/OmsAgent/README.md)

Download links removed from README.md. This change is up for discussion; currently those links must be updated with each new release, which should be avoided since [onboard_agent.sh](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/onboard_agent.sh) was created.

### CODE REVIEW

**Comment by lagalbra** [comment]
Updated with Azure CLI 2.0 deployment/list commands.

**Comment by keikhara** [comment]
This doc should add the new azure cli 2.0 section. New azure cli will only support ARM. 
https://docs.microsoft.com/en-us/cli/azure/vm/extension

**Comment by lagalbra** [comment]
@keikhara Can you approve if you want me to merge this?

File: docs/VM-Extension.md [hunk 1]
```diff
@@ -176,24 +227,27 @@ Set-AzureRmVMExtension -ResourceGroupName $RGName -VMName $VmName -Location $Loc
...
-- Oracle Linux 5,6, and 7 (x86/x64)
-- Red Hat Enterprise Linux Server 5,6 and 7 (x86/x64)
-- Debian GNU/Linux 6, 7, and 8 (x86/x64)
-- Ubuntu 12.04 LTS, 14.04 LTS, 15.04 (x86/x64)
-- SUSE Linux Enterprise Server 11 and 12 (x86/x64)
+* CentOS Linux 5,6, and 7 (x86/x64)
+* Oracle Linux 5,6, and 7 (x86/x64)
+* Red Hat Enterprise Linux Server 5,6 and 7 (x86/x64)
+* Debian GNU/Linux 6, 7, and 8 (x86/x64)
+* Ubuntu 12.04 LTS, 14.04 LTS, 15.04, 15.10, 16.04 LTS (x86/x64)
```
Comment (NarineM) [inline docs/VM-Extension.md]: Please add Ubuntu16.10 to supported list.

File: docs/VM-Extension.md [hunk 2]
```diff
@@ -69,16 +85,42 @@ You can change to Azure Resource Manager mode by running:
...
 
-You can deploy the OmsAgent Extension by running:
+You can deploy the OMS Agent for Linux Extension by running:
 ```
-$ azure vm extension set <resource-group> <vm-name> \
-OmsAgentForLinux Microsoft.EnterpriseCloud.Monitoring <version> \
---public-config-path public.json  \
---private-config-path protected.json
+$ azure vm extension set <resource-group> \
+                         <vm-name> \
```
Comment (NarineM) [inline docs/VM-Extension.md]: all vm extension examples already present in azure-vm extension docs. Can you update this doc to link them instead?

**Comment by lagalbra** [comment]
Closing; please refer to PR https://github.com/Microsoft/OMS-Agent-for-Linux/pull/504

**Pull request closed without merging**